File: models/product.model.js
const moongoose = require('mongoose');

const productSchema = new moongoose.Schema({
    name: {
        type: String,
        required: true
    },
    price: {
        type: Number,
        required: true
    },
    category: {
        type: String,
        required: true

    },
    stock: {
        type: Number
    }
}, { timestamps: true });

module.exports = moongoose.model('Product', productSchema);

File: FastAPI/e-commerce/app/models/product_model.py
from typing import Optional
from pydantic import BaseModel

class Product(BaseModel):
    name: str = None
    price: float
    category: str
    stock: int

class ProductPatch(BaseModel):
    name: Optional[str] = None
    price: Optional[float] = None
    category: Optional[str] = None
    stock: Optional[int] = None